pkg/kvstore/etcd.go

```
@@ -81,6 +81,9 @@ func newEtcdClient(config *client.Config, cfgPath string) (KVClient, error) {
 		}
 	}
 	if config != nil {
+		if config.DialTimeout == 0 {
+			config.DialTimeout = 10 * time.Second
```

I think 1 minute would be better. Why not retrying forever instead of bailing out after 10 seconds?

---

1 minute is fine. 

Let's say a user forgets to start etcd.  Unless the user checks that etcd is
around they might not notice the agent is waiting for them to do something.
Failing hard after a timeout as the default behaviour is friendlier IMO.

Please let me know if I misunderstood https://github.com/cilium/cilium/issues/1234

Thanks.

---

Failing is good. It will cause the pod to be restarted with an error condition which will indicate to the user that something is not OK. If we retry forever there is no indication to the user that something is going wrong.